coopdevs/timeoverflow
# Session, cookies and security

### What is this?
We found out that people tend to leave their TO account logged in while using public computers. This is a security concern, of course.

We use cookies to store sessions, this is not the best solution since cookies can be stolen and we should switch to some database store for sessions (Memcache, Redis, etc). But that's out of the scope of this PR.

With this PR we're addressing:

1. Update Devise gem to get some important security update
2. Adopt the new Rails `secrets.yml` and remove a deprecation warning, more info [here](http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#config-secrets-yml)
3. Add an expiration to the session cookie

### How to test it
1. Test that you must log in again after 1 hour you stopped using TO
2. Test that if you check the option `Remember me` during login it shouldn't ask to log in again during 2 weeks

### GIF
![](https://media.giphy.com/media/l9vF1aRG7d4J2/giphy.gif)
